command-parser/src/parser.rs

```diff
@@ -112,6 +112,55 @@ impl<'a> Parser<'a> {
         })
     }
 
+    /// Parses a command out of a buffer.
+    ///
+    /// Instead of using the list of set prefixes can you here define a
+    /// function that returns the index at which the prefix ends if it
+    /// matches, or return none if there is no prefix that matches.
```

Comment:
Instead of using the list of set prefixes, define a closure that manually tests for a prefix. The closure must return an option, with the Some value being the first index of the name of the command.